ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -1719,14 +1735,40 @@ FEDataManager::interpWeighted(const int f_data_idx,
 
                     dof_id_scratch = F_dof_indices[i];
                     F_dof_map.constrain_element_vector(F_rhs, dof_id_scratch);
-                    F_vec.add_vector(F_rhs, dof_id_scratch);
+                    if (is_ghosted)
+                    {
+                        for (unsigned int i = 0; i < dof_id_scratch.size(); ++i)
+                        {
+                            F_local_soln[F_petsc_vec->map_global_to_local_index(dof_id_scratch[i])]
```

Comment:
Can we avoid this lookup for DOF ids that are owned by the processor?

Reply:
(We would still need to do the lookup for DOF ids owned by other processors, to find where those entries are stored in the ghost region.)

Reply:
Yes and no. If you look at the code for `map_global_to_local_index` it starts with something like

```cpp
if (proc_dof_start <= dof && dof < proc_dof_end)
    return dof - proc_dof_start;
```
so the translation is cheap for on-processor dofs: libMesh stores a `map` for dofs in the ghost region which is more expensive to use.

We could avoid it by inlining that check ourselves (which I think the compiler may actually do) but I would prefer not to do it since we are going to cache the indexing anyway in #513.

Reply:
Oh yeah, now I remember seeing this when I was looking at the libMesh code. I agree there is no need for us also to have that check here.